### DISCUSSION TITLE
context stack with more lines

### DISCUSSION THREAD

**Discussion by TommyJerryMairo** [hugsy/gef#786] [Q&A]
Is it possible to have something like `context stack --size=35` which means GEF will temporarily change the `nb_lines_stack` to `35` when executing the command? The default 8 lines output fits my screen when the context is printed while hitting a breakpoint. However, sometimes I might want to inspect the whole stack frame which requires more lines to print. Currently, I'm using a user-defined command to change `nb_lines_stack` to `35`, run the context stack, and then change the  `nb_lines_stack` back to `8`. This is ugly as both `35` and `8` are hardcoded. Also, I noticed that `context stack` is not autocompleted by GEF while `context` itself is, which means I could use `conte[tab]` to complete `context`. However, `context st[tab]` has no response from GEF.

**Comment by Grazfather** [comment]
That's a pretty good idea.

**Reply by CaledoniaProject** [reply]
Is it implemented?

**Reply by CaledoniaProject** [reply]
In gdb-peda, I can just do `stack 50` instead of `ctx stack`

**Reply by hugsy** [reply]
The stack display is nothing more than the `dereference` command applies to `$sp`. So yes, it's been implemented and documented literally for years:

```
gef➤  dereference -l 50 $sp
```

After that feel free to create an alias.

**Comment by hugsy** [comment]
GEF tries to offer a maximum of flexibility, because we simply can't handle each individual preference, it allows to do it yourself in a (hopefully) seamless manner.

For what you mention, you can easily create a new pane to the `context` and register it using [`register_external_context_pane`](https://hugsy.github.io/gef/api/gef/#function-register_external_context_pane) function to customize how the stack is shown. Just like I mentioned above, there is no `stack` command: it simply (by default) dereferences the stack pointer. If you want a custom layout for the stack you can easily do so as such:

```python
def mystack_content(): 
  gdb.execute("dereference -l 35 $sp") # or whatever length you prefer

def mystack_title(): 
  return "mystack" 

register_external_context_pane("mystack", mystack_context, mystack_title)
```

Then configure how you want the context layout to be shown (search docs for `gef config context.layout`). 

Hope that answers.

**Reply by CaledoniaProject** [reply]
Can I add this in ~/.gdbinit?

**Reply by hugsy** [reply]
Yes

**Reply by CaledoniaProject** [reply]
I fixed several typos and here's my ~/.gdbinit file

```
source ~/.gdbinit-gef.py
set follow-fork-mode parent

python
def mystack_content():
    gdb.execute("dereference -l 35 $sp")

def mystack_title():
    return "mystack"

register_external_context_pane("mystack", mystack_content, mystack_title)

end
```

But there's no mystack command in gef, what's wrong?

**Reply by hugsy** [reply]
Use `gef config context.layout`
